crates/volta-core/src/tool/mod.rs

```diff
@@ -192,3 +193,27 @@ fn registry_fetch_error(
     let from_url = from_url.as_ref().to_string();
     || ErrorKind::RegistryFetchError { tool, from_url }
 }
+
+pub fn check_shim_reachable(shim_name: &str) {
+    let home = match volta_home() {
+        Ok(home) => home,
+        Err(_) => return,
+    };
+
+    let shim = home.shim_file(shim_name);
+    let resolved = match which::which(shim_name) {
+        Ok(resolved) => resolved,
+        Err(_) => {
+            info!(
+                "{} cannot find command {}. Please ensure that {} is at the front of your PATH.",
+                note_prefix(),
+                shim_name,
+                shim.display()
+            );
+            return;
+        }
+    };
+    if resolved != shim {
+        info!("{} {} is shadowed by another binary of the same name at {}. Please ensure that {} is at the front of your PATH.", note_prefix(), shim_name, resolved.display(), shim.display());
```

Comment:
Similar comments as above for this message re `shim_dir` and `PATH`/`Path`. I also wonder if it would make sense to expand the CTA somewhat, to make it clear why we're suggesting it? Perhaps something like:

```
To ensure your commands work as expected, please move {} to the start of your PATH.
```